File: aihpc/kernel/cuda/vector_add/vector_add.py
'''
Author       : Thinksky5124
Date         : 2024-04-07 23:51:14
LastEditors  : Thinksky5124
LastEditTime : 2024-04-07 23:56:54
Description  : file content
FilePath     : /AIHPC-Larning/aihpc/kernel/cuda/vector_add/vector_add.py
'''
import torch
from aihpc.core import Dispatcher, BackendType, LaunchKernel

def add(x: torch.Tensor, y: torch.Tensor):
    
    return x + y

@Dispatcher.register(BackendType.CUDA, "add")
class LaunchCUDAAddKernel(LaunchKernel):
    def __call__(self, *args: torch.Any, **kwargs: torch.Any) -> torch.Any:
        return add(*args, **kwargs)

File: aihpc/kernel/cuda/vector_add/vector_add.cpp
/*
 * @Author       : Thinksky5124
 * @Date         : 2024-04-18 15:29:12
 * @LastEditors  : Thinksky5124
 * @LastEditTime : 2024-04-20 16:12:29
 * @Description  : file content
 * @FilePath     : /AIHPC-Larning/aihpc/kernel/cuda/vector_add/vector_add.cpp
 */
#include <torch/all.h>
#include <pybind11/pybind11.h>
#include "vector_add_cuda.cuh"
#include "vector_add.h"

namespace kernel
{
    namespace cuda
    {
        void vector_add()
        {
            launch_vector_add_kernel();
        }
    } // namespace cuda
    
} // namespace kernel